modules/ocf_backups/files/prune-old-backups

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import json
+import shlex
+import subprocess
+import sys
+import textwrap
+
+RETENTION_TIME = datetime.timedelta(days=180) # 6 months
+
+BOX_FTP = 'ftps://ftp.box.com:990'
+BACKUP_PREFIX = 'ocf-backup-'
+
+LFTP_INIT = '''
+    set ftps:initial-prot ""
+    set ftp:ssl-force true
+    set ftp:ssl-protect-data true
+    open {box_ftp}
+    user {email} {passwd}
+'''
+
+def parse_date(dtstr):
+    """Converts a 'YYYY-MM-DD' formatted string to a datetime.date object."""
+    if len(dtstr) != 10:
+        raise ValueError
+
+    year = int(dtstr[0:4])
+
+    if dtstr[4] != '-':
+        raise ValueError
+
+    month = int(dtstr[5:7])
+
+    if dtstr[7] != '-':
+        raise ValueError
+
+    day = int(dtstr[8:10])
+
+    return datetime.date(year, month, day)
+
+def list_items(creds):
+    """List all items at the root of the Box.com folder."""
+    lftp_cmd = (LFTP_INIT + 'renlist').format(
+        box_ftp=BOX_FTP,
+        email=creds['email'],
+        passwd=creds['password'],
+    )
+
+    p = subprocess.run(
+        ['lftp', '--norc'],
+        input=lftp_cmd.encode(),
+        stdout=subprocess.PIPE,
+    )
+
+    p.check_returncode()
+
+    return p.stdout.decode().splitlines()
+
+def delete_items(creds, items, quiet):
+    lftp_cmd = (LFTP_INIT + 'rm -r {items}').format(
+        box_ftp=BOX_FTP,
+        email=creds['email'],
+        passwd=creds['password'],
+        items=' '.join(shlex.quote(item) for item in items)
+    )
+
+    if quiet:
+        stdout_pipe = subprocess.DEVNULL
+    else:
+        stdout_pipe = None
+
+    p = subprocess.run(
+        ['lftp', '--norc'],
+        input=lftp_cmd.encode(),
+        stdout=stdout_pipe
+    ).check_returncode()
+
+def is_old_backup(filename):
+    """Returns true if this backup is old enough to be deleted."""
+    start, _, end = filename.partition(BACKUP_PREFIX)
+
+    # Not a backup
+    if start:
+        return False
+
+    try:
+        backup_date = parse_date(end)
+    except ValueError:
+        return False
+
+    return datetime.date.today() - backup_date > RETENTION_TIME
+
+def main(args):
+    with open('/opt/share/backups/box-creds.json') as f:
```

Comment:
put this at the top as `CREDS_PATH` or something?

Reply:
Sure, we should probably do that for the `upload-to-box` script as well.